micro/example_handler_test.go

```diff
@@ -30,11 +30,14 @@ func ExampleHandler() {
 	rec := rectangle{10, 5}
 
 	config := micro.Config{
-		Name:    "RectangleAreaService",
-		Version: "0.1.0",
-		Endpoint: micro.Endpoint{
-			Handler: rec,
-			Subject: "rectangle.area",
+		Name:        "RectangleAreaService",
+		Version:     "0.1.0",
+		RootSubject: "area",
+		Endpoints: map[string]micro.Endpoint{
+			"Rectangle": {
+				Subject: "rec",
```

Comment:
Whats the benefit of having this be configurable rather than `RootSubject.EndpointName`? We should specify valid end point names matching kv bucket rules so they are always safe to use y concatenation.  

We could make this a optional but I think it would need to have significant benefit in being setable (and can always later be added, but not removed)

Reply:
I'm all for it, using endpoint name as subject may be a better option, especially since calling the field `Subject` here is a bit misleading (it's not the actual subject, just the part after root).

One thing I'm not sure about is validating the endpoints the same way as kv bucket names. From what I remember, bucket names cannot contain subject delimiters, whereas for endpoints I would imagine users wanting to have logical hierarchy, similarly to REST urls, e.g. `foo.users.create` and `foo.users.delete`, where `foo` is the root subject`.

Reply:
@aricart curious about your thoughts here

Reply:
I think that endpoint subject == endpoint name.
The Endpoint subject should be ... a valid subject, so we don't care if users wants to have endpoints with `.` or not.
As we enocurage users to put some queryable tokens in subjects, removing that possibility from services does not seem right.

We might in future consider different approach to build service subjects hierarchy though...

HTTP framerowsk often allow grouping:
```
r := http.Router()
v1 := r.Group("/v1")
products := v1.Group("products")
products.Endpoint("add", handler)
```

Which is useful for structuring and adding group specific logic/middleware etc.

Reply:
> @aricart curious about your thoughts here

I think that if we take the api that @Jarema is pointing above, the need for the root subject etc goes away - the group becomes a root subject for that set of endpoints. In otherwords, I think both should be supported - easy should be trivial, and grouped should be easy.